accounts/views.py

```diff
@@ -43,13 +45,27 @@ class HomeView(ListView):
     template_name = "accounts/home.html"
     context_object_name = "tweets"
     model = Tweet
-    queryset = Tweet.objects.select_related("user").order_by("-created_at")
+    queryset = (
+        Tweet.objects.select_related("user")
+        .order_by("-created_at")
+        .prefetch_related("like_set")
+        .annotate(like_count=Count("like"))
+    )
 
     def get_context_data(self, **kwargs):
         ctx = super().get_context_data(**kwargs)
+        liked_list = []
+        tweets = Tweet.objects.select_related("user").prefetch_related("like_set").all()
+        for tweet in tweets:
+            liked_list.extend(
+                tweet.like_set.filter(user=self.request.user).values_list(
+                    "tweet_id", flat=True
+                )
+            )

```

Comment:
for文で回すとクエリの数増えるので、基本このような書き方はしないでください。
```suggestion
        ctx["liked_list"] = Like.objects.filter(user=self.request.user).values_list(
            "tweet", flat=True
        )
```
このようにしてくだされば大丈夫です

Reply:
修正しました。